packages/react-topology/src/components/TopologyView/examples/TopologyContextMenuDemo.tsx

```diff
@@ -0,0 +1,201 @@
+      </g>
+    </DefaultNode>
+  );
+};
+
+const customLayoutFactory: LayoutFactory = (type: string, graph: Graph): Layout | undefined => {
+  switch (type) {
+    case 'Cola':
+      return new ColaLayout(graph);
+    default:
+      return new ColaLayout(graph, { layoutOnDrag: false });
+  }
+};
+
+const customComponentFactory: ComponentFactory = (kind: ModelKind, type: string) => {
+  switch (type) {
+    case 'group':
+      return DefaultGroup;
+    default:
+      switch (kind) {
+        case ModelKind.graph:
+          return GraphComponent;
+        case ModelKind.node:
+          return withContextMenu(() => contextMenu)(CustomNode);
+        case ModelKind.edge:
+          return DefaultEdge;
+        default:
+          return undefined;
+      }
+  }
+};
+
+const NODE_DIAMETER = 75;
+
+const NODES: NodeModel[] = [
+  {
+    id: 'node-0',
+    type: 'node',
+    label: 'Node 0',
+    width: NODE_DIAMETER,
+    height: NODE_DIAMETER,
+    shape: NodeShape.ellipse
+  },
+  {
+    id: 'node-1',
+    type: 'node',
+    label: 'Node 1',
+    width: NODE_DIAMETER,
+    height: NODE_DIAMETER,
+    shape: NodeShape.hexagon
+  },
+  {
+    id: 'node-2',
+    type: 'node',
+    label: 'Node 2',
+    width: NODE_DIAMETER,
+    height: NODE_DIAMETER,
+    shape: NodeShape.octagon
+  },
+  {
+    id: 'node-3',
+    type: 'node',
+    label: 'Node 3',
+    width: NODE_DIAMETER,
+    height: NODE_DIAMETER,
+    shape: NodeShape.rhombus
+  },
+  {
+    id: 'node-4',
+    type: 'node',
+    label: 'Node 4',
+    width: NODE_DIAMETER,
+    height: NODE_DIAMETER,
+    shape: NodeShape.hexagon
+  },
+  {
+    id: 'node-5',
+    type: 'node',
+    label: 'Node 5',
+    width: NODE_DIAMETER,
+    height: NODE_DIAMETER,
+    shape: NodeShape.rect
+  },
+  {
+    id: 'Group-1',
+    children: ['node-0', 'node-1', 'node-2'],
+    type: 'group',
+    group: true,
+    label: 'Group-1',
+    style: {
+      padding: 40
+    }
+  }
+];
+
+const EDGES = [
+  {
+    id: 'edge-node-4-node-5',
+    type: 'edge',
+    source: 'node-4',
+    target: 'node-5',
+    edgeStyle: EdgeStyle.default
+  },
+  {
+    id: 'edge-node-0-node-2',
+    type: 'edge',
+    source: 'node-0',
+    target: 'node-2',
+    edgeStyle: EdgeStyle.default
+  }
+];
+
+const contextMenuItem = (label: string, i: number): React.ReactElement => {
+  if (label === '-') {
+    return <ContextMenuSeparator key={`separator:${i.toString()}`} />;
+  }
+  return (
+    // eslint-disable-next-line no-alert
+    <ContextMenuItem key={label} onClick={() => alert(`Selected: ${label}`)}>
+      {label}
+    </ContextMenuItem>
+  );
+};
+
+const createContextMenuItems = (...labels: string[]): React.ReactElement[] => labels.map(contextMenuItem);
+
+const contextMenu = createContextMenuItems('First', 'Second', 'Third', '-', 'Fourth');
+
+export const TopologyContextMenuDemo: React.FC = () => {
+  const [selectedIds, setSelectedIds] = React.useState<string[]>([]);
```

Comment:
No need for selections (nodes and edges aren't using withSelection) in this example